store/tikv/coprocessor.go

```diff
@@ -495,9 +507,28 @@ func (worker *copIteratorWorker) run(ctx context.Context) {
 		close(task.respChan)
 		select {
 		case <-worker.finishCh:
+			worker.actionOnExceed.mu.Lock()
+			worker.actionOnExceed.mu.aliveWorker--
+			worker.actionOnExceed.mu.Unlock()
 			return
 		default:
 		}
+
+		worker.actionOnExceed.mu.Lock()
+		if worker.actionOnExceed.mu.exceeded != 0 {
+			logutil.BgLogger().Info("memory exceeds quota, end one copIterator worker.",
+				zap.String("copIteratorWorker id ", worker.id))
+
+			worker.actionOnExceed.mu.aliveWorker--
+			// reset action
+			worker.actionOnExceed.mu.exceeded = 0
+			worker.actionOnExceed.once = sync.Once{}
+
+			worker.actionOnExceed.mu.Unlock()
+			// end this worker
+			return
+		}
+		worker.actionOnExceed.mu.Unlock()
```

Comment:
defer unlock

Reply:
We run this logic in a `for-range` scope. so I tend to manually handle the lock instead of using `defer`.

Reply:
defer unlock after Line:517.